docs/upgrading.md

```diff
@@ -10,6 +10,29 @@ We're building on the version `v3.4.0` a forward compatibility layer to help use
 To help on the migration process, all deprecated components are being marked with `@deprecated` and deprecated behaviour will trigger a `E_USER_DEPRECATED` error.
 However, you can also find here the instructions on how to make your code compatible with both versions.
 
+### General migration strategy
+
+It is recommended to update your existing software from any previous 3.x version to the latest 3.4.x version using 
```

Comment:
Isn't this implied on the mention of the compatibility layer above? Perhaps we don't need a sub-section for the "General migration strategy" but add a paragraph above encouraging people to go through v3.4?

The remarks about PHPUnit are important but maybe using a note/warning might get more attention?